lfs/pointer.go

```diff
@@ -17,27 +18,41 @@ var (
 		"https://hawser.github.com/spec/v1",  // pre-release
 		"https://git-lfs.github.com/spec/v1", // public launch
 	}
-	latest = "https://git-lfs.github.com/spec/v1"
-
-	oidType  = "sha256"
-	oidRE    = regexp.MustCompile(`\A[0-9a-fA-F]{64}`)
-	template = `version %s
-oid sha256:%s
-size %d
-`
+	latest      = "https://git-lfs.github.com/spec/v1"
+	oidType     = "sha256"
+	oidRE       = regexp.MustCompile(`\A[[:alnum:]]{64}`)
 	matcherRE   = regexp.MustCompile("git-media|hawser|git-lfs")
+	extRE       = regexp.MustCompile(`\Aext-\d{1}-\w+`)
 	pointerKeys = []string{"version", "oid", "size"}
 )
 
 type Pointer struct {
-	Version string
-	Oid     string
-	Size    int64
-	OidType string
+	Version    string
+	Oid        string
+	Size       int64
+	OidType    string
+	Extensions []PointerExtension
+}
+
+type PointerExtension struct {
+	Name     string
+	Priority int
+	Oid      string
+	OidType  string
 }
 
-func NewPointer(oid string, size int64) *Pointer {
-	return &Pointer{latest, oid, size, oidType}
+type ByPriority []PointerExtension
+
+func (p ByPriority) Len() int           { return len(p) }
+func (p ByPriority) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p ByPriority) Less(i, j int) bool { return p[i].Priority < p[j].Priority }
+
+func NewPointer(oid string, size int64, exts []PointerExtension) *Pointer {
```

Comment:
I think this should use `*PointerExtension` where possible, so you don't have to worry about what functions take the direct objects, and which take pointers.